pip/commands/show.py

```diff
@@ -12,6 +13,13 @@
 logger = logging.getLogger(__name__)
 
 
+_normalize_re = re.compile(r'[^a-z0-9-.]', re.I)
+
+
+def normalize_name(name):
+    return _normalize_re.sub('-', name.lower())
+
+
```

Comment:
Indeed, my bad.
install is in fact using `pkg_resources.safe_name(package_name)` I think